lib/game.rb

```diff
@@ -0,0 +1,134 @@
+class Game
+  attr_accessor :board,:player_1,:player_2
+  attr_reader :winner
+  WIN_COMBINATIONS = [
+    [0,1,2], 
+    [3,4,5],  
+    [6,7,8], 
+    [0,3,6],
+    [1,4,7],
+    [2,5,8],
+    [0,4,8],
+    [6,4,2],
+  ]
+    def initialize(player_1=Human.new("X"),player_2=Human.new("O"),board=Board.new)
+	    @player_1 = player_1
+	    @player_2 = player_2
+	    @board = board
+    end
+
+
+  def current_player
+    if @board.turn_count % 2 == 0
+      @player_1
+    else
+      @player_2
+    end
+  end
+
+
+  def over?
+    if @board.full? || won? || draw?
+    	true
+  end
+end
+
+
+ def won?
+  @winner = nil
+        WIN_COMBINATIONS.each do |wincombo|
+          if wincombo.all? {|position| @board.cells[position] == "X"}
+            @winner = "X"
+          elsif wincombo.all? {|position| @board.cells[position] == "O"}
+            @winner = "O"
+          end
+        end
+        if @winner
+          return true
+        else
+          false
+        end
+  end
+
+  def winner
+    if won?
+      @winner
+    end
```

Comment:
Could also be written as `@winner if won?` if you would like a single-line approach.